noodles-bed/src/de.rs

```diff
@@ -0,0 +1,934 @@
+use std::fmt;
+use std::io::BufRead;
+use std::ops::{AddAssign, MulAssign, Neg};
+
+use noodles_core::Position;
+
+use serde::Deserialize;
+use serde::de::{
+    self, DeserializeSeed, EnumAccess, IntoDeserializer, MapAccess, SeqAccess,
+    VariantAccess, Visitor,
+};
+
+use crate::{Record, reader::Reader, error::{Error, Result}};
```

Comment:
`bed::Record` can be changed in `record.rs`.
https://github.com/umccr/noodles/blob/07bd3d0b984157e39e3f09c2b40996f1f1725ff4/noodles-bed/src/record.rs#L111-L116

For example, to ignore the two extra structs, use the [flatten](https://serde.rs/field-attrs.html#flatten) field attribute. This means that during serialisation/deserialisation, `standard_fields` and `optional_fields` won't be present. See the example [here](https://serde.rs/attr-flatten.html).
```rust
/// A BED record.
#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
pub struct Record<const N: u8> {
    #[serde(flatten)]
    standard_fields: StandardFields,
    #[serde(flatten)]
    optional_fields: OptionalFields,
}
```